model/src/resource.rs

```diff
@@ -208,6 +208,10 @@ impl CrdExt for Resource {
     fn object_meta(&self) -> &ObjectMeta {
         self.meta()
     }
+
+    fn to_yaml(&self) -> Result<String, serde_yaml::Error> {
+        serde_yaml::to_string(self)
+    }
```

Comment:
You should be able to move this to the body of to_yaml in `crd_ext`.

Reply:
That was my initial thought, but using `serde_yaml::to_string(self)` in `crd_ext` gives an error that `self` does not implement `Serialize`. Since `CrdExt` is a trait and not a struct, I wasn't sure if I could even make it derive `Serialize`.

Reply:
You can define the trait as `pub trait CrdExt: Serialize` to force everything that uses the trait to be serializable.